catalyst_voices/lib/widgets/rich_text/voices_rich_text.dart

```diff
@@ -0,0 +1,259 @@
+                          iconData: CatalystVoicesIcons.rt_decrease_indent,
+                        ),
+                        controller: _controller,
+                        isIncrease: false,
+                      ),
+                      QuillToolbarImageButton(
+                        options: const QuillToolbarImageButtonOptions(
+                          iconData: CatalystVoicesIcons.photograph,
+                        ),
+                        controller: _controller,
+                      ),
+                    ],
+                  ),
+                ),
+              ),
+            ),
+          Padding(
+            padding: const EdgeInsets.symmetric(horizontal: 16.0),
+            child: ResizableBoxParent(
+              minHeight: 400,
+              resizable: true,
+              child: Container(
+                decoration: BoxDecoration(
+                  color: _editMode
+                      ? theme.colors.onSurfaceNeutralOpaqueLv1
+                      : theme.scaffoldBackgroundColor,
+                  border: Border.all(
+                    color: theme.colorScheme.outlineVariant,
+                  ),
+                  borderRadius: BorderRadius.circular(8),
+                ),
+                child: QuillEditor.basic(
+                  controller: _controller,
+                  focusNode: _focusNode,
+                  configurations: QuillEditorConfigurations(
+                    padding: const EdgeInsets.all(16),
+                    placeholder: 'Start writing your text...', //TODO localize
+                    embedBuilders: isWeb()
+                        ? FlutterQuillEmbeds.editorWebBuilders()
+                        : FlutterQuillEmbeds.editorBuilders(),
+                  ),
+                ),
+              ),
+            ),
+          ),
+          if (widget.charsLimit != null)
+            Padding(
+              padding: const EdgeInsets.symmetric(horizontal: 32.0),
+              child: Row(
+                children: [
+                  Expanded(
+                    child: Text(
+                      'Supporting text', //TODO localize
+                      style: theme.textTheme.bodySmall,
+                    ),
+                  ),
+                  Text(
+                    '${_documentLength}/${widget.charsLimit!}',
+                    style: theme.textTheme.bodySmall,
+                  ),
+                ],
+              ),
+            ),
+          SizedBox(height: 16),
+          Container(
+            padding: const EdgeInsets.symmetric(
+              horizontal: 16.0,
+              vertical: 12.0,
+            ),
+            alignment: Alignment.centerRight,
+            color: theme.colors.onSurfaceNeutralOpaqueLv1,
+            child: VoicesFilledButton(
+              child: Text('Save'.toUpperCase()), //TODO localize
+              onTap: () => widget.onSave?.call(_controller.document),
+            ),
+          ),
```

Comment:
This `Save` button probably should be out of this widget

Reply:
I double checked with Niels, he is sure that they will be always used together.